pages/api/auth/signin.js

```diff
@@ -0,0 +1,45 @@
+import bcrypt from "bcrypt";
+import sql from "~/lib/postgres";
+import * as yup from "yup";
+import jwt from "jsonwebtoken";
+
+const schema = yup.object().shape({
+  email: yup.string().email().required(),
+  hash_password: yup.string().required(),
+});
+
+const handler = async ({ method, body: { email, password } }, res) => {
+  if (method === "POST") {
+    const user = {
+      email: email,
+      hash_password: password,
+    };
+    const schemaValid = await schema.isValid(user);
+    if (schemaValid) {
+      const userExists = await sql`SELECT * FROM users WHERE email=${email}`;
+      if (userExists.count > 0) {
+        try {
+          const userValid = await bcrypt.compare(
+            password,
+            userExists[0].hash_password
+          );
+          if (userValid) {
+            return res.status(200).json({ message: "Success" });
```

Comment:
If we're logging a user in, we might want to return that user's data, so we can say something like `Welcome back, Johnny`